samza-yarn/src/main/java/org/apache/samza/validation/YarnJobValidationTool.java

```diff
@@ -158,23 +159,24 @@ public void validateJmxMetrics() throws Exception {
     CoordinatorStreamStore coordinatorStreamStore = new CoordinatorStreamStore(config, metricsRegistry);
     coordinatorStreamStore.init();
     try {
-      Config configFromCoordinatorStream = CoordinatorStreamUtil.readConfigFromCoordinatorStream(coordinatorStreamStore);
-      ChangelogStreamManager changelogStreamManager = new ChangelogStreamManager(coordinatorStreamStore);
-      JobModelManager jobModelManager =
-          JobModelManager.apply(configFromCoordinatorStream, changelogStreamManager.readPartitionMapping(),
-              coordinatorStreamStore, metricsRegistry);
+      LocalityManager localityManager =
+          new LocalityManager(new NamespaceAwareCoordinatorStreamStore(coordinatorStreamStore, SetConfig.TYPE));
       validator.init(config);
-      Map<String, String> jmxUrls = jobModelManager.jobModel().getAllContainerToHostValues(SetContainerHostMapping.JMX_TUNNELING_URL_KEY);
-      for (Map.Entry<String, String> entry : jmxUrls.entrySet()) {
-        String containerId = entry.getKey();
-        String jmxUrl = entry.getValue();
-        log.info("validate container " + containerId + " metrics with JMX: " + jmxUrl);
-        JmxMetricsAccessor jmxMetrics = new JmxMetricsAccessor(jmxUrl);
-        jmxMetrics.connect();
-        validator.validate(jmxMetrics);
-        jmxMetrics.close();
-        log.info("validate container " + containerId + " successfully");
+      LocalityModel localityModel = localityManager.readLocality();
+
+      for (ContainerLocality containerLocality : localityModel.getContainerLocalities().values()) {
+        String containerId = containerLocality.id();
+        String jmxUrl = containerLocality.jmxTunnelingUrl();
+        if (StringUtils.isNotBlank(jmxUrl)) {
```

Comment:
In samza we control JMX enable disable with a config, you should only emit JMX metrics here if that config is enabled

Reply:
this is a CLI tool and I kept the old semantics as is in terms of how it uses the locality information. I'd prefer to keep as is since this PR's scope is not to fix how yarn job validation tool uses JMX.